### DISCUSSION TITLE
Include the contents of a source file (.py, .c, etc...) inside a codeblock.

### DISCUSSION THREAD

**Discussion by Adam-Antios** [quarto-dev/quarto-cli#4077] [Q&A] [answered]
It would be very useful to be able to include the contents of a source file using the command

    {{< include file_name.py >}}

directly into a code block in order to be rendered as code, with proper formatting and code highlighting.

**Comment by cderv** [comment, marked as answer]
I believe this can be done today using this extension https://github.com/quarto-ext/include-code-files that you can add to your project

Did you try already ? 

Also related feature request: https://github.com/quarto-dev/quarto-cli/issues/2314

**Reply by Adam-Antios** [reply]
You are right, this does exactly that. Thank you very much. Generally, I think this would be really useful as a core feature of Quarto.
